--- origsrc/gtk-engines-0.12/metal/Makefile.am	1999-05-30 03:45:05.000000000 -0500
+++ src/gtk-engines-0.12/metal/Makefile.am	2007-12-16 13:03:05.250000000 -0600
@@ -11,6 +11,6 @@
 	metal_theme_draw.c	\
 	metal_theme_main.c
 
-libmetal_la_LDFLAGS = -module -avoid-version
+libmetal_la_LDFLAGS = -module -avoid-version -no-undefined
 libmetal_la_LIBADD = $(GTK_LIBS)
 
--- origsrc/gtk-engines-0.12/notif/Makefile.am	1999-05-30 03:45:08.000000000 -0500
+++ src/gtk-engines-0.12/notif/Makefile.am	2007-12-16 13:03:08.890625000 -0600
@@ -9,6 +9,6 @@
 	notif_theme_draw.c	\
 	notif_theme_main.c
 
-libnotif_la_LDFLAGS = -module -avoid-version
+libnotif_la_LDFLAGS = -module -avoid-version -no-undefined
 libnotif_la_LIBADD = $(GTK_LIBS)
 
--- origsrc/gtk-engines-0.12/pixmap/Makefile.am	1999-02-02 08:25:04.000000000 -0600
+++ src/gtk-engines-0.12/pixmap/Makefile.am	2007-12-16 13:03:12.390625000 -0600
@@ -11,5 +11,5 @@
 	pixmap_theme_main.c	\
 	pixmap_theme.h
 
-libpixmap_la_LDFLAGS = -module -avoid-version
+libpixmap_la_LDFLAGS = -module -avoid-version -no-undefined
 libpixmap_la_LIBADD = $(GDK_IMLIB_LIBS)
--- origsrc/gtk-engines-0.12/raleigh/Makefile.am	2000-11-21 17:37:02.000000000 -0600
+++ src/gtk-engines-0.12/raleigh/Makefile.am	2007-12-16 13:03:15.796875000 -0600
@@ -10,6 +10,6 @@
 	raleigh_theme_draw.c	\
 	raleigh_theme_main.c
 
-libraleigh_la_LDFLAGS = -module -avoid-version
+libraleigh_la_LDFLAGS = -module -avoid-version -no-undefined
 libraleigh_la_LIBADD = $(GTK_LIBS)
 
--- origsrc/gtk-engines-0.12/redmond95/Makefile.am	1999-05-30 03:45:11.000000000 -0500
+++ src/gtk-engines-0.12/redmond95/Makefile.am	2007-12-16 13:03:19.421875000 -0600
@@ -10,6 +10,6 @@
 	redmond95_theme_draw.c	\
 	redmond95_theme_main.c
 
-libredmond95_la_LDFLAGS = -module -avoid-version
+libredmond95_la_LDFLAGS = -module -avoid-version -no-undefined
 libredmond95_la_LIBADD = $(GTK_LIBS)
 
